client/my-sites/media/details.jsx

```diff
@@ -0,0 +1,247 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import page from 'page';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import { EditorMediaModalDetail } from 'post-editor/media-modal/detail';
+import { getMimePrefix, getMimeType } from 'lib/media/utils';
+import ImageEditor from 'blocks/image-editor';
+import VideoEditor from 'blocks/video-editor';
+import MediaActions from 'lib/media/actions';
+
+import MediaLibrarySelectedStore from 'lib/media/library-selected-store';
+import accept from 'lib/accept';
+
+class MediaDetails extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+		selected: PropTypes.number,
+		items: PropTypes.array,
+		view: PropTypes.string,
+	};
+
+	closeDetailsModal = () => {
+		page( '/media/' + this.props.site.slug + '/' );
+		return;
+	};
+
+	onEditorCancel = () => {
+		page( '/media/' + this.props.site.slug + '/' + this.getSelectedMedia().ID );
+		return;
+	};
+
+	onImageEditorDone = ( error, blob, imageEditorProps ) => {
+		if ( error ) {
+			this.onEditImageCancel( imageEditorProps );
+			return;
+		}
+
+		const { fileName, site, ID, resetAllImageEditorState } = imageEditorProps;
+
+		const mimeType = getMimeType( fileName );
+
+		const item = {
+			ID: ID,
+			media: {
+				fileName: fileName,
+				fileContents: blob,
+				mimeType: mimeType,
+			},
+		};
+
+		MediaActions.update( site.ID, item, true );
```

Comment:
I'm really hesitant to introduce new features or functionality that uses the legacy media flux store. Unfortunately, only part of the store has been migrated to use Redux at this time, which is quite a blocker, unless you are willing to tackle it as well 😉 

Could we reuse the contents of the existing dialog (`EditorMediaModalDetail`, `ImageEditor`, `VideoEditor`),  as much as possible? This would reduce the effort needed to migrate the remaining bits of the media section to Redux.

This might require some componentization to allow easier reuse, as I've pointed out in other comments in this review.